osu.Game/Online/SocketClient.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace osu.Game.Online
+{
+    public abstract class SocketClient : IAsyncDisposable
+    {
+        public event Func<Exception?, Task>? Closed;
```

Comment:
Would be good to have xmldoc explaining this, since it's a pretty complex delegate structure.